x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/controller.go

```diff
@@ -105,25 +121,41 @@ func (e *Controller) Update(c *config.Config) error {
 	e.ast = rawAst
 	e.lock.Unlock()
 
-	return e.update()
+	rErr = e.update(ctx)
+	return
 }
 
 // Set sets the transpiler vars for dynamic inputs resolution.
-func (e *Controller) Set(vars []*transpiler.Vars) {
+func (e *Controller) Set(ctx context.Context, vars []*transpiler.Vars) {
+	var err error
+	span, ctx := apm.StartSpan(ctx, "Set", "app.internal")
+	defer func() {
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
+		span.End()
+	}()
 	e.lock.Lock()
 	ast := e.ast
 	e.vars = vars
 	e.lock.Unlock()
 
 	if ast != nil {
-		err := e.update()
+		err = e.update(ctx)
 		if err != nil {
 			e.logger.Errorf("Failed to render configuration with latest context from composable controller: %s", err)
 		}
 	}
 }
 
-func (e *Controller) update() error {
+func (e *Controller) update(ctx context.Context) (err error) {
+	span, ctx := apm.StartSpan(ctx, "update", "app.internal")
+	defer func() {
+		if err != nil {
+			apm.CaptureError(ctx, err).Send()
+		}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, err).Send()
```